core/src/main/java/org/elasticsearch/action/support/TransportActions.java

```diff
@@ -31,16 +31,9 @@
 
     public static boolean isShardNotAvailableException(Throwable t) {
         Throwable actual = ExceptionsHelper.unwrapCause(t);
-        if (actual instanceof IllegalIndexShardStateException) {
-            return true;
-        }
-        if (actual instanceof IndexMissingException) {
-            return true;
-        }
-        if (actual instanceof IndexShardMissingException) {
-            return true;
-        }
-        if (actual instanceof NoShardAvailableActionException) {
+        if (actual instanceof ShardNotFoundException ||
```

Comment:
we miss catching  the replacement of IndexMissingHere.. either IndexNotFoundException or ResourceNotFoundException (making ShardNotFoundException redundant)